test/e2e/network/proxy.go

```diff
@@ -258,9 +262,83 @@ var _ = SIGDescribe("Proxy", func() {
 				framework.Failf(strings.Join(errs, "\n"))
 			}
 		})
+
+		ginkgo.It("proxy connection returns a series of 301 redirections for a pod", func() {
+			ns := f.Namespace.Name
+			httpVerbs := []string{"DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"}
```

Comment:
We should not add tests for 301 redirects of DELETE/PATCH/POST/PUT, and should fix the API server to stop sending 301 redirects for those verbs, since it will not work properly with any known clients.

We also should not add tests for 301 redirects for an OPTIONS request, and should fix the server to return an Allow header indicating an OPTIONS, GET, or HEAD request may be made, not a redirect.

Testing 301 for GET and HEAD is ok.